src/openforms/variables/utils.py

```diff
@@ -34,26 +38,18 @@ def check_initial_value(initial_value: JSONObject, data_type: str) -> JSONObject
         return DEFAULT_INITIAL_VALUE[data_type]
 
 
-def get_variables_for_context(submission: "Submission") -> dict[str, JSONValue]:
-    """Return the key/value pairs of static variables and submission variables.
-
-    This returns the saved component variables and user defined variables for a submission, plus the static variables
-    (which are never saved). Note that depending on when it is called, the 'auth' static variables
-    (auth_bsn, auth_kvk...) can be already hashed.
+def get_variables_for_context(submission: Submission) -> dict[str, VariableValue]:
     """
-    from openforms.formio.service import FormioData
+    Return the key/value pairs of data in the state (static, user-defined, and component
+    variables). If specified in the settings, strings will be escaped for HTML.
 
-    from .service import get_static_variables
+    Note that depending on when it is called, the 'auth' static variables (auth_bsn,
+    auth_kvk...) can be already hashed.
+    """
+    state = submission.load_submission_value_variables_state()
+    data = state.get_data(include_static_variables=True).data
 
-    data = submission.data.data
     if settings.ESCAPE_REGISTRATION_OUTPUT:
-        data = html_escape_json(data)
-
-    formio_data = FormioData(
-        **{
-            variable.key: variable.initial_value
-            for variable in get_static_variables(submission=submission)
-        },
-        **data,
-    )
```

Comment:
we have a behaviour change here - the static variables are no longer overwritten by submission data. I'm okay with it, but we need to document this like [here](https://open-forms.readthedocs.io/en/3.2.0/changelog.html#upgrade-procedure). Can you add a notice already in the changelog so we don't forget?